lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -130,5 +99,72 @@ def clean(self):
         if not usage_key:
             return
 
-        cleaned_data['user'] = self.clean_requested_user(cleaned_data, usage_key.course_key)
+        cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
+
+    def _clean_requested_user(self, cleaned_data, course_key):
+        """
+        Validates and returns the requested_user, while checking permissions.
+        """
+        requesting_user = self.initial['requesting_user']
+        requested_username = cleaned_data.get('username', None)
+
+        if not requested_username:
```

Comment:
yay!  I'm all for readability.  And thanks for your comments in making it even better.